tools/ci_build/github/azure-pipelines/linux-gpu-tensorrt-ci-pipeline.yml

```diff
@@ -14,7 +14,7 @@ jobs:
     ALLOW_RELEASED_ONNX_OPSET_ONLY: '1'
   workspace:
     clean: all
-  pool: Linux-CPU-2019
+  pool: onnxruntime-tensorrt-linuxbuild-T4 
```

Comment:
You will start to see longer waiting time on this pipeline, since we only have very few T4 machines available. And I couldn't get any new machine. So, we still should split them. And, from security & compliance part of view, we also need to do it. Because we should minimize build environment dependencies. For example. ONNX is not a build time dependency of ORT, But if you change this pipeline back, it becomes a dependency and will pull a lot more in.

Reply:
Minimizing dependencies is documented in Biden's Cyber EO. You can find the full text of order at https://www.whitehouse.gov/briefing-room/presidential-actions/2021/05/12/executive-order-on-improving-the-nations-cybersecurity/ .  The order asks: "documenting and minimizing dependencies on enterprise products that are part of the environments used to develop, build, and edit software." For the dependencies we can't remove, we will pay much time on documenting them and upgrading them. That's why I suggest splitting build and test . Software in test environment could not have any impact on the bits we ship, so we can exclude them from the compliance requirements.

Reply:
Understand your concerns. @jywu-msft, what is your though on this?

Reply:
> Understand your concerns. @jywu-msft, what is your though on this?

I think that splitting build/test in separate phases is probably a good thing. @chilo-ms, this is why I mentioned in our discussion offline that we could refactor to keep the testing as separate phase but support docker as well (so we can test in either Ubuntu or Centos containers)
@snnn , rolling back is a short term way to move forward asap since we want to get back the previous behavior of testing using docker.